R/simulation-set.R

```diff
@@ -55,39 +55,45 @@ SimulationSet <- R6::R6Class(
       # Test and validate the simulation objects
       validateIsString(simulationSetName)
       validateIsString(simulationFile)
-      validateIsString(massBalanceFile, nullAllowed = TRUE)
       validateIsFileExtension(simulationFile, "pkml")
+      validateIsUnitFromDimension(timeUnit, "Time")
+      
+      # For optional input, usually null is allowed
+      # but not here as it would mean that nothing would be reported
+      validateIsIncluded(c(applicationRanges), ApplicationRanges)
+      validateIsString(massBalanceFile, nullAllowed = TRUE)
       if (!isEmpty(massBalanceFile)) {
         validateIsFileExtension(massBalanceFile, "json")
         self$massBalanceSettings <- jsonlite::fromJSON(massBalanceFile, simplifyVector = FALSE)[["MassBalancePlots"]]
       }
-      validateIsString(timeUnit, nullAllowed = TRUE)
-      validateIsPositive(object = minimumSimulationEndTime, nullAllowed = TRUE)
-      validateIsNumeric(timeOffset)
-      # For optional input, usually null is allowed
-      # but not here as it would mean that nothing would be reported
-      validateIsIncluded(c(applicationRanges), ApplicationRanges)
-
+      
       # Before loading the simulation, check if the file exists
       validateFileExists(simulationFile)
       simulation <- ospsuite::loadSimulation(simulationFile, addToCache = FALSE)
-
+      # Checks requiring simulation info
+      endTime <- ospsuite::toUnit(
+        quantityOrDimension = "Time",
+        values = simulation$outputSchema$endTime,
+        targetUnit = timeUnit
+      )
+      validateVectorRange(timeOffset, type = "numeric", valueRange = c(0, endTime))
```

Comment:
Question: if the `minimumSimulationEndTime` is passed: is the simulation end time adjusted **before** or **after** this check? Because if it's done after the range bound should be `max(simulation$outputSchema$endTime, minimumSimulationEndTime)`

Reply:
Oups, before.
Also, I noticed I forced `minimumSimulationEndTime` to be within 0 and `outputSchema$endTime`...
What would be the expected range for this argument actually ? `>0`, `[0, endTime]`, or `[endTime, Inf]`

Reply:
I think it should be `>0`
Also if `minimumSimulationEndTime <= endTime` I actually would not change the simulation end time.